### PR TITLE
Add 'data' in PluginException in Cisco Umbrella Investigate plugin

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
Add 'data' in PluginException in Cisco Umbrella Investigate plugin

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting..\nrapid7/Cisco Umbrella Investigate:3.1.1. Step name: categorization\n",
    "meta": {},
    "output": {
      "categories": [
        {
          "content_categories": [
            "Research/Reference"
          ],
          "name": "example.com",
          "security_categories": [],
          "status": 0
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_investigate:3.1.1 --debug run < tests/categorization.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: Tier 1 does not have access to bulk endpoints\n",
    "log": "Connect: Connecting..\nrapid7/Cisco Umbrella Investigate:3.1.1. Step name: categorization\nAn error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: Tier 1 does not have access to bulk endpoints\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/actions/categorization/action.py\", line 22, in run\n    remoteCategories = self.connection.investigate.categorization(domains, labels=True)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 161, in categorization\n    return self._post_categorization(domains, labels)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 148, in _post_categorization\n    json.dumps(domains)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 136, in post_parse\n    return self._request_parse(self.post, uri, params, data)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 95, in _request_parse\n    data=response.text\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: Tier 1 does not have access to bulk endpoints\n\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 311, in handle_step\n    output = self.start_step(input_message['body'], 'action', logger, log_stream, is_test, is_debug)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 419, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/actions/categorization/action.py\", line 24, in run\n    raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: Tier 1 does not have access to bulk endpoints\n\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_investigate:3.1.1 --debug run < tests/categorization_multiple_domains_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: {\"error\":\"unauthorized\"}\n",
    "log": "Connect: Connecting..\nrapid7/Cisco Umbrella Investigate:3.1.1. Step name: categorization\nAn error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: {\"error\":\"unauthorized\"}\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/actions/categorization/action.py\", line 22, in run\n    remoteCategories = self.connection.investigate.categorization(domains, labels=True)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 159, in categorization\n    return self._get_categorization(domains, labels)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 141, in _get_categorization\n    return self.get_parse(uri, params)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 130, in get_parse\n    return self._request_parse(self.get, uri, params)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/investigate/investigate.py\", line 95, in _request_parse\n    data=response.text\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: {\"error\":\"unauthorized\"}\n\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 311, in handle_step\n    output = self.start_step(input_message['body'], 'action', logger, log_stream, is_test, is_debug)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 419, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.7/site-packages/cisco_umbrella_investigate_rapid7_plugin-3.1.1-py3.7.egg/komand_cisco_umbrella_investigate/actions/categorization/action.py\", line 24, in run\n    raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)\nkomand.exceptions.PluginException: An error occurred during plugin execution!\n\nSomething unexpected occurred. Check the logs and if the issue persists please contact support. Response was: An error occurred during plugin execution!\n\nUnauthorized. Request had Authorization header but token was missing or invalid. Please ensure your API token is valid. Response was: {\"error\":\"unauthorized\"}\n\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/cisco_umbrella_investigate:3.1.1 --debug run < tests/categorization_bad.json
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "EncodingValidator" failed!
        Cause: A forbidden character was found in the 'description' field of the spec.yaml file: The SHA256 checksum of the sample. This checksum is important if you’d like to find out more about this sample in the /sample/ endpoint

Validator "ExampleInputValidator" failed!
        Cause: All inputs should have example field in plugin.spec.
In action "search": input "include_category", there is no example field.


----
[*] Total time elapsed: 2562.215ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
help.md:783: MD014 Dollar signs used before commands without showing output
help.md:818: MD014 Dollar signs used before commands without showing output
help.md:974: MD014 Dollar signs used before commands without showing output
help.md:1046: MD014 Dollar signs used before commands without showing output
help.md:1155: MD014 Dollar signs used before commands without showing output

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
[FAIL] Fails markdown linting

[*] Validating python files for style...
./komand_cisco_umbrella_investigate/actions/categorization/action.py:20:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/domain_tags/action.py:31:150: E202 whitespace before '}'
./komand_cisco_umbrella_investigate/actions/ns_whois/action.py:18:12: E111 indentation is not a multiple of four
./komand_cisco_umbrella_investigate/actions/sample/action.py:19:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample/action.py:22:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/samples/action.py:20:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/samples/action.py:23:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/samples/action.py:26:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample_artifacts/action.py:19:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample_artifacts/action.py:22:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample_connections/action.py:19:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample_connections/action.py:22:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample_samples/action.py:19:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/sample_samples/action.py:22:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/search/action.py:22:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/search/action.py:25:1: W293 blank line contains whitespace
./komand_cisco_umbrella_investigate/actions/search/action.py:26:53: E711 comparison to None should be 'if cond is None:'
./komand_cisco_umbrella_investigate/connection/connection.py:6:1: F403 'from ..investigate import *' used; unable to detect undefined names
./komand_cisco_umbrella_investigate/connection/connection.py:14:9: E265 block comment should start with '# '
./komand_cisco_umbrella_investigate/connection/connection.py:19:28: F405 'investigate' may be undefined, or defined from star imports: ..investigate
./komand_cisco_umbrella_investigate/connection/connection.py:23:19: F405 're' may be undefined, or defined from star imports: ..investigate
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
69 [0.. 50.. ]
Run started:2021-02-13 17:28:12.501596

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 3353
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[FAIL] Fails Misspell check
assessment.md:203:74: "occurences" is a misspelling of "occurrences" assessment.md:203:194: "occurences" is a misspelling of "occurrences" assessment.md:203:314: "occurences" is a misspelling of "occurrences" komand_cisco_umbrella_investigate/actions/cooccurrences/action.py:11:39: "occurences" is a misspelling of "occurrences" komand_cisco_umbrella_investigate/actions/cooccurrences/action.py:25:39: "occurences" is a misspelling of "occurrences" komand_cisco_umbrella_investigate/actions/cooccurrences/action.py:28:32: "occurences" is a misspelling of "occurrences"
```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "EncodingValidator" failed!
        Cause: A forbidden character was found in the 'description' field of the spec.yaml file: The SHA256 checksum of the sample. This checksum is important if you’d like to find out more about this sample in the /sample/ endpoint

Validator "ExampleInputValidator" failed!
        Cause: All inputs should have example field in plugin.spec.
In action "search": input "include_category", there is no example field.


----
[*] Total time elapsed: 79571.09ms
```
<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: cisco_umbrella_investigate/help.md [hunk 1]
```diff
@@ -1425,6 +1425,7 @@ This plugin does not contain any troubleshooting information.
 
 # Version History
 
+* 3.1.1 - Add `data` in PluginException error handling in `investigate.py` file
```
Comment (jschipp-r7) [inline cisco_umbrella_investigate/help.md:1428]:
```suggestion
* 3.1.1 - Return information from API when a PluginException has occurred
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
@r7-kszczepanskagorna Can you check for a unicode char in the description?

```
Validator "EncodingValidator" failed!
        Cause: A forbidden character was found in the 'description' field of the spec.yaml file
```

**Review by jschipp-r7 — APPROVED** [review]
